serializer/serializer.c

```diff
@@ -75,6 +75,62 @@ static status_t ta_hash243_queue_to_json_array(hash243_queue_t queue,
   return SC_OK;
 }
 
+static status_t ta_json_array_to_hash8019_array(cJSON const* const obj,
+                                                char const* const obj_name,
+                                                hash8019_array_p array) {
+  status_t ret = SC_OK;
+  flex_trit_t hash[FLEX_TRIT_SIZE_8019] = {};
+  cJSON* json_item = cJSON_GetObjectItemCaseSensitive(obj, obj_name);
+  if (cJSON_IsArray(json_item)) {
+    cJSON* current_obj = NULL;
+    cJSON_ArrayForEach(current_obj, json_item) {
+      if (current_obj->valuestring != NULL) {
+        flex_trits_from_trytes(hash, NUM_TRITS_SERIALIZED_TRANSACTION,
+                               (tryte_t const*)current_obj->valuestring,
+                               NUM_TRYTES_SERIALIZED_TRANSACTION,
+                               NUM_TRYTES_SERIALIZED_TRANSACTION);
+        hash_array_push(array, hash);
+      }
+    }
+  } else {
+    return SC_CCLIENT_JSON_PARSE;
```

Comment:
If you move the condition `!cJSON_IsArray(json_item)` earlier, you can shorten the code.

That is,
```c
if (! cJSON_IsArray(json_item))
    return SC_CCLIENT_JSON_PARSE;

cJSON* current_obj = NULL;
cJSON_ArrayForEach(current_obj, json_item) {
    if (current_obj->valuestring != NULL) {
        flex_trits_from_trytes(hash, NUM_TRITS_SERIALIZED_TRANSACTION,
                               (tryte_t const*)current_obj->valuestring,
                               NUM_TRYTES_SERIALIZED_TRANSACTION,
                               NUM_TRYTES_SERIALIZED_TRANSACTION);
        hash_array_push(array, hash);
      }
}
```